src/main/java/shef/data/Recipe.java

```diff
@@ -61,6 +64,16 @@ public Recipe(String name, String description, Set<String> tags, Set<String> ing
     this.spinOffs = new HashSet<>();
   }
 
+  /** Creates a Recipe from a Datastore entity. */
+  public Recipe(Entity recipeEntity) {
+    this.name = (String) recipeEntity.getProperty("name");
```

Comment:
Are you still going to be able to write tests that involve Recipes with this constructor? You'd have to make a mock Entity object, but I don't know how you would do that.

Reply:
I think I can just instantiate an Entity and then set its properties using entity.setProperty()